windows/amd64: unable to debug minidumps created via GOTRACEBACK=wer

Getting `error while reading spliced memory at ...: EOF` when attempting to debug a minidump created using the `GOTRACEBACK=wer` feature from Go 1.21+

1. What version of Delve are you using (`dlv version`)?
```
Delve Debugger
Version: 1.23.1
Build: $Id: 2eba762d75437d380e48fc42213853f13aa2904d $
```
2. What version of Go are you using? (`go version`)?
```
go version go1.23.2 windows/amd64
```
3. What operating system and processor architecture are you using?
```
Windows 11 (23H2 22631.4169) 
amd64
```
4. What did you do?
* Create a minidump by running a crashing application under `GOTRACEBACK=wer`
* Debug the minidump with `dlv core <exe> <dmp>` 
* Try to print stack trace, goroutines, etc
5. What did you expect to see?
* Expect to see stack trace, goroutines, etc. 
* `dlv core` works if I run the program under `dlv debug ...`, then within dlv run `dump myprog.dmp`, then finally use `dlv core`
6. What did you see instead?
```
(dlv) bt
0  0x00007ffde8af0612 in ???
   at ?:-1
   error: error while reading spliced memory at 0x9: EOF
(truncated)
(dlv) grs
Command failed: error while reading spliced memory at 0x603418: EOF
```


### Short reproducible example

(First, enable local dumps collection by creating reg key at `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\Windows Error Reporting\LocalDumps`
See https://learn.microsoft.com/en-us/windows/win32/wer/collecting-user-mode-dumps for more info)
Then:
```
> type main.go
package main

import "fmt"

func main() {
        var somethingnil *string
        fmt.Println("on no", *somethingnil)
}

> go build main.go
> set GOTRACEBACK=wer
> .\main.exe
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0xb69614]
...
> dlv core .\main.exe %LOCALAPPDATA%\CrashDumps\main.exe.23592.dmp
Type 'help' for list of commands.
(dlv) bt
0  0x00007ffde8af0612 in ???
   at ?:-1
   error: error while reading spliced memory at 0x9: EOF
```

--- a/pkg/proc/stack.go
+++ b/pkg/proc/stack.go
@@ -382,15 +382,31 @@ func (it *stackIterator) stacktrace(depth int) ([]Stackframe, error) {
 		return nil, errors.New("negative maximum stack depth")
 	}
 	frames := make([]Stackframe, 0, depth+1)
-	it.stacktraceFunc(func(frame Stackframe) bool {
+	f := func(frame Stackframe) bool {
 		frames = append(frames, frame)
 		return len(frames) < depth+1
-	})
+	}
+	it.stacktraceFunc(f)
+	if it.Err() != nil && len(frames) == 1 && it.g != nil && frames[0].SystemStack && (it.opts&StacktraceSimple == 0) {
+		// If we can't continue from the first frame, and it was on a system stack
+		// and we have a goroutine which we are allowed to switch to then switch
+		// to it and continue the stacktrace from there.
+		// This improves stacktraces produced on Windows by WER where the first
+		// thread will be executing a system function from which we can't continue
+		// to trace.
+		// See #3824.
+		it.err = nil
+		it.opts |= StacktraceG
+		it.stacktraceFunc(f)
+	}
+
 	if err := it.Err(); err != nil {
 		if len(frames) == 0 {
 			return nil, err
 		}
+
 		frames = append(frames, Stackframe{Err: err})
+
 	}
 	return frames, nil
 }

